utils/stm.stanza

```diff
@@ -0,0 +1,302 @@
+      var recommended-voltage-data
+
+      match(recommended-volt-data) :
+        (recommended-volt-data: JObject) :
+          val min-volts = match(recommended-volt-data["min"]) :
+            (min-volts-double:Double) : min-volts-double
+            (min-volts) : fatal("Invalid recommended-voltage min data")
+          val max-volts = match(recommended-volt-data["max"]) :
+            (max-volts-double:Double) : max-volts-double
+            (max-volts) : fatal("Invalid recommended-voltage max data")
+          val nominal = match(recommended-volt-data["nominal"]) :
+            (nominal:True) : nominal
+            (nominal:False) : nominal
+            (nominal) : fatal("Invalid nominal data")
+          recommended-voltage-data = STMVoltageLimits(min-volts, max-volts, nominal)
+        (recommended-volt-data) :
+          fatal("Invalid JSON") 
+
+      stm-pow-pin = STMPowerPin(name, recommended-voltage-data)
+
+  val pin-properties = Vector<STMPinProperties-Row>()
+  val pins = pin-prop["pins"]
+  match(pins) :
+    (pins:Tuple<JObject>) :
+      for pin-item in pins do :
+        val pin-name = match(pin-item["pin"]) :
+          (pin-name:String) : pin-name
+          (pin-name) : fatal("Invalid Pin Name")
+        val pad-name = match(pin-item["pad"]) :
+          (pad-double:Double) : pad-double
+          (pad-str:String) : pad-str
+          (pad-data) : fatal("Invalid Pad Value")
+        val side = match(pin-item["side"]) :
+          (side-val:String) : 
+            switch(side-val) :
+              "Up" : Up,
+              "Down" : Down,
+              "Left" : Left,
+              "Right" : Right
+          (side-val) : fatal("Invalid Pin Name")
+        val power-props? = match(pin-item["power-pin?"]) :
+          (power-val:True) : power-val
+          (power-val:False) : power-val
+          (power-val) : fatal("Invalid Power Value")
+        val generic-props? = match(pin-item["generic-pin?"]) :
+          (generic-val:True) : generic-val
+          (generic-val:False) : generic-val
+          (generic-val) : fatal("Invalid Generic Value")
+
+        add(pin-properties, STMPinProperties-Row(pin-name, pad-name, side, generic-props?, power-props?))
+    (pins) :
+      fatal("Invalid JSON")
+ 
+  val stm-properties = STMPinProperties(stm-gen-pin, stm-pow-pin, to-tuple(pin-properties))
+  stm-properties
+
+;=================================================
+;==================== Testing ====================
+;=================================================
+
+; val JSON-TEST-FILE-LIST = to-hashset<String> $ [
+;   "stm-test.json"
+; ]
+
+; defn main ():
+;   for file in JSON-TEST-FILE-LIST do :
+;     val input-json = parse $ slurp(file)
+;     val pin-properties = create-pin-properties(input-json)
+;     val bundles = create-bundles(input-json)
+;     val supports = create-supports(input-json)
+;     println(pin-properties)
+;     println(bundles)
+;     println(supports)
+; main()
```

Comment:
To be moved to a test file in jitpcb/tests/

Reply:
Delete commented out code